### PR TITLE
forgejo-runner: 4.0.1 -> 5.0.3, various quality-of-life improvements

### PR DESCRIPTION
**Pull request by christoph-heiss**
Changelog: https://code.forgejo.org/forgejo/runner/src/branch/main/RELEASE-NOTES.md#5.0.3 (and following)

Also includes some quality-of-life improvements to the package itself:
- enable running package tests by only disabling those that require network access
- add nix-update-script passthru for automated updates
- use (new-ish) `versionCheckHook` for running the `--version` check

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by getchoo — CHANGES_REQUESTED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 360535`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>forgejo-runner</li>
  </ul>
</details>

File: pkgs/by-name/fo/forgejo-runner/package.nix [hunk 1]
```diff
@@ -1,47 +1,63 @@
...
 
-  passthru.tests = {
-    inherit (nixosTests.forgejo) sqlite3;
-    version = testers.testVersion {
-      package = forgejo-runner;
-      version = src.rev;
-    };
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.sqlite3 = nixosTests.forgejo.sqlite3;
```
[the remark this answers is not in the document]
Comment (christoph-heiss) [reply pkgs/by-name/fo/forgejo-runner/package.nix]: Yep, seems sensible, thanks! :+1:

**Review by getchoo — APPROVED** [review]

**Review by emilylange — APPROVED** [review]
Hi, thanks, this looks very good.

I am personally a bit short on time and not all that interested in Forgejo Actions anymore, so feel free to consider yourself the main maintainer of forgejo-runner if you want :)

Oh and Kranzes has not reacted to forgejo-runner PRs/updates/etc. in a long time, just as a fyi.

**Comment by christoph-heiss** [comment]
> Hi, thanks, this looks very good.
> 
> I am personally a bit short on time and not all that interested in Forgejo Actions anymore, so feel free to consider yourself the main maintainer of forgejo-runner if you want :)
> 
> Oh and Kranzes has not reacted to forgejo-runner PRs/updates/etc. in a long time, just as a fyi.

Alright, thanks for the info!

**Pull request merged**